pkg/controller/externalsecrets_controller.go

```diff
@@ -40,23 +74,295 @@ type ExternalSecretsReconciler struct {
+		r.log.V(4).Info("received reconcile event", "object", fmt.Sprintf("%T", obj), "name", obj.GetName(), "namespace", obj.GetNamespace())
+
+		objLabels := obj.GetLabels()
+		if objLabels != nil {
+			// will look for custom label set on objects not created in external-secrets namespace, and if it exists,
+			// namespace in the reconcile request will be set same, else since label check matches is an object
+			// created by controller, and we safely assume, it's in the external-secrets namespace.
+			namespace := objLabels[externalsecretsNamespaceMappingLabelName]
+			if namespace == "" {
+				namespace = obj.GetNamespace()
+			}
+
+			labelOk := func() bool {
+				if objLabels[requestEnqueueLabelKey] == requestEnqueueLabelValue {
+					return true
+				}
+				value := objLabels[externalsecretsResourceWatchLabelName]
+				if value == "" {
+					return false
+				}
+				key := strings.Split(value, "_")
+				if len(key) != 2 {
+					r.log.Error(fmt.Errorf("invalid label format"), "%s label value(%s) not in expected format on %s resource", externalsecretsResourceWatchLabelName, value, obj.GetName())
+					return false
+				}
+				namespace = key[0]
+				return true
+			}
+
+			if labelOk() && namespace != "" {
+				return []reconcile.Request{
+					{
+						NamespacedName: types.NamespacedName{
+							Name:      externalsecretsObjectName,
+							Namespace: namespace,
+						},
+					},
+				}
+			}
+		}
+
+		r.log.V(4).Info("object not of interest, ignoring reconcile event", "object", fmt.Sprintf("%T", obj), "name", obj.GetName(), "namespace", obj.GetNamespace())
+		return []reconcile.Request{}
+	}
+
+	// predicate function to ignore events for objects not managed by controller.
+	controllerManagedResources := predicate.NewPredicateFuncs(func(object client.Object) bool {
+		return object.GetLabels() != nil && object.GetLabels()[requestEnqueueLabelKey] == requestEnqueueLabelValue
+	})
+
+	// predicate function to filter events for objects which controller is interested in, but
+	// not managed or created by controller.
+	controllerWatchResources := predicate.NewPredicateFuncs(func(object client.Object) bool {
+		return object.GetLabels() != nil && object.GetLabels()[externalsecretsResourceWatchLabelName] != ""
+	})
+
+	withIgnoreStatusUpdatePredicates := builder.WithPredicates(predicate.GenerationChangedPredicate{}, controllerManagedResources)
+	controllerWatchResourcePredicates := builder.WithPredicates(predicate.ResourceVersionChangedPredicate{}, controllerWatchResources)
+	controllerManagedResourcePredicates := builder.WithPredicates(controllerManagedResources)
+
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&operatorv1alpha1.ExternalSecrets{}).
+		For(&operatorv1alpha1.ExternalSecretsOperator{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		Named(ControllerName).
+		Watches(&certmanagerv1.Certificate{}, handler.EnqueueRequestsFromMapFunc(mapFunc), withIgnoreStatusUpdatePredicates).
```

Comment:
you can loop here (for the multiple `.Watches(..)`) as well if you really want to reduce lines – 
helps keep the WatchResources under one single source of truth, avoiding us to cause silly errors of mentioning a resource here/above.

Or, you can protect them via a test so we enforce all known resources are always watched.